infrastructure/src/main/java/org/corfudb/infrastructure/management/FailureDetector.java

```diff
@@ -174,13 +173,22 @@ private PollReport pollRound(long epoch,
                 // Timeout is increased anytime a failure is encountered.
                 // Failure includes both, unresponsive and outOfPhaseEpoch nodes.
                 if (failuresDetected) {
+                    final long iterationElapsedTime = TimeUnit.MILLISECONDS.convert(
+                            System.nanoTime() - iterationStartTime, TimeUnit.NANOSECONDS);
+
+                    // Increase the inter iteration sleep time if the period time increases
+                    interIterationSleepTime = Math.max(interIterationInterval,
+                                                       period - iterationElapsedTime);
                     period = getIncreasedPeriod();
                     tuneRoutersResponseTimeout(members, routerMap, new HashSet<>(members), period);
                 }
                 final int pollIteration = iteration;
                 responses.forEach(s -> responsesMap.put(s, pollIteration));
             }
-            Sleep.MILLISECONDS.sleepUninterruptibly(interIterationInterval);
+            log.debug("Time taken for iteration's failure detection: {}",
+                    TimeUnit.MILLISECONDS.convert(System.nanoTime() - iterationStartTime,
+                            TimeUnit.NANOSECONDS));
+            Sleep.MILLISECONDS.sleepUninterruptibly(interIterationSleepTime);
```

Comment:
Why sleep again here, if we tune the timeouts on every iteration for pollOnceAsync, shouldn't the timeouts on those requests also reflect interIterationSleepTime?

Reply:
The sleep at the end of each iteration is for poll iteration intervals when a failure observed and separate from response timeout. changed variable names and added comments to better describe it in the code.